src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -758,6 +768,29 @@ class PostgresEntityDataQueryService(
         return WriteEvent(System.currentTimeMillis(), numUpdates)
     }
 
+    private fun lockEntitiesForUpdate(connection: Connection, entityKeyIds: Set<UUID>, partitionsInfo: PartitionsInfo) {
+        // Acquire entity key id locks
+        lockEntitiesForUpdate(
+                connection,
+                PostgresArrays.createUuidArray(connection, entityKeyIds),
+                PostgresArrays.createIntArray(connection, partitionsInfo.partitions),
```

Comment:
Shouldn't we also filter partitions by entity key ids?
`entityKeyIds.map { getPartition(it, partitionsInfo .partitions) }`

Reply:
+1 updated for this